### PR TITLE
Register and Login features

### PR DESCRIPTION
**Pull request by JayKay24**
**What the PR does.**
Adds register and login features

**PR Description**
Implement Register and Login functionalities in the application.

**How to test manually**

Clone the repository at https://github.com/JayKay24/bucket_list_app.git
Register with the application. This will take you to the login route.
Log in to use the application. This will take you to the homepage.
Change directory into tests folder: python -m test_registration

**Expected Outcome**
All tests passed

### CODE REVIEW

**Comment by Mbarak-Mbigo** [comment]
Do you think someone who has never interacted with your code can use your README and this PR summary information to successfully test the changes you have implemented?

File: application/config.py [hunk 1]
```diff
@@ -8,6 +8,6 @@ class Configuration(object): # Instruct Flask to run the application in DEBUG mo
     """
     DEBUG = True,
     SECRET_KEY = 'development key'
```
Comment (Mbarak-Mbigo) [inline application/config.py]: Is this key you have provided meet the requirements a given key should meet?

File: application/forms.py [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import wtforms
 from wtforms import validators
-from classes.user import User
```
Comment (Mbarak-Mbigo) [inline application/forms.py]: Separate your imports into three categories: *standard*, *third party* and *local imports*

File: tests/test_bucket_list.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import unittest
-from application.classes.bucket_list import BucketList
+from application.models.bucket_list import BucketList
 
 class BucketListTest(unittest.TestCase):
     def setUp(self):
```
Comment (Mbarak-Mbigo) [inline tests/test_bucket_list.py]: Whenever you define *setUp*, it's highly encouraged to define *tearDown* in order to clear what you set up

**Review by Mbarak-Mbigo — COMMENTED** [review]
*Format Imports appropriately*, 
*Ensure you add comprehensive error handling* especially on edge cases

File: tests/test_bucket_list.py (hunk 3 above)
Comment (JayKay24) [reply tests/test_bucket_list.py]: @Mbarak-Mbigo please check my PR message again. I will implement the recommended changes.

**Comment by AwuorMarvin** [comment]
Looks good man. Took some tips from your LF.
